imports/api/HandScoreCalculator.js

```diff
@@ -0,0 +1,151 @@
+import Constants from './Constants';
+import { GameRecordUtils } from './utils/GameRecordUtils';
+
+export var HandScoreCalculator = {
+    jpn: {
+        dealin_delta: jpn_dealin_delta,
+        selfdraw_delta: jpn_selfdraw_delta,
+        mistake_delta: jpn_mistake_delta
+    }
+};
+
+/**
+ * Calculates the change in points as a result of a deal-in hand
+ * Uses the formula from:
+ * https://en.wikipedia.org/wiki/Japanese_Mahjong_scoring_rules
+ *
+ * @param {Number} points - the total number of points in hand
+ * @param {Number} fu - the fu value of the hand
+ * @param {Number} bonus - the current round bonus
+ * @param {string} dealerWind - the true wind of the current dealer
+ * @param {string} winnerWind - the winning seat true wind
+ * @param {string} loserWind - the losing seat true wind
+ * @param {Number} riichiSticks - the number of riichi sticks won
+ * @return {Object} containing the point difference for each seat
+ */
+function jpn_dealin_delta(points,
+                          fu,
+                          bonus,
+                          dealerWind,
+                          winnerWind,
+                          loserWind,
+                          riichiSticks) {
+    let winds = {};
+    Constants.WINDS.forEach(w => {winds[w] = 0});
+
+    let handValue;
+    let bonusPoints = bonus * Constants.JPN_BONUS_POINTS;
+
+    // The multiplier is for whether or not it's a dealer victory
+    let multiplier = (winnerWind != dealerWind) ? 4 : 6;
+
+    // Check to see if you have to count basic points
+    if (points < 5) {
+        if (fu === 20 || (points === 1 && fu === 25)) {
+            throw TypeError("Invalid points/fu combination");
```

Comment:
Maybe we can make our own custom exception for this, since TypeError isn't exactly correct

Reply:
You're right, it should be RangeError